pkg/cloud/services/networking/router.go

```diff
@@ -45,17 +45,26 @@ func (s *Service) ReconcileRouter(openStackCluster *infrav1.OpenStackCluster, cl
 	}
 
 	routerName := getRouterName(clusterName)
-	s.scope.Logger().Info("Reconciling router", "name", routerName)
+	routerListOpts := routers.ListOpts{Name: routerName}
+	existingRouter := false
+	if openStackCluster.Spec.Router != nil {
+		routerListOpts = openStackCluster.Spec.Router.ToListOpt()
+		existingRouter = true
+	} else {
+		s.scope.Logger().Info("Reconciling router", "name", routerName)
```

Comment:
Can you move this back up to the top so it's unconditional? I would like to see this log message even when we're going to re-use an existing router: it's a useful marker.

Reply:
We can add a log at the top, but without specifying the name of the router. Like this
s.scope.Logger().Info("Reconciling router", "cluster", clusterName)
Or we can display the log after receiving the real name of the router by filter. What will be better?

Reply:
I still think the first option is better. Since the message with the real id and the name of the router is already below
`s.scope.Logger().V(6).Info(fmt.Sprintf("Reuse existing Router %s with id %s", router.Name, router.ID))`